src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/OpBuilder.java

```diff
@@ -422,6 +426,26 @@ private static FuncOp createExternTypeHelperFunc(Map<ExternalizedTypeElement, Li
 
     FuncOp build(String name, Op op) {
         Value ancestorBody = builder.op(constant(type(Body.Builder.class), null));
+        // check if java version at compile time matches the java version at runtime
```

Comment:
Now we have the right code shape we could move this to a static helper method rather that repeating the logic.